awsshell/utils.py

```diff
@@ -22,6 +26,25 @@ def build_config_file_path(file_name):
     return os.path.join(os.path.expanduser('~'), '.aws', 'shell', file_name)
 
 
+@contextlib.contextmanager
+def temporary_file(mode):
+    """Cross platform temporary file creation.
+
+    This is an alternative to ``tempfile.NamedTemporaryFile`` that
+    also works on windows and avoids the "file being used by
+    another process" error.
+    """
+    tempdir = tempfile.mkdtemp()
+    basename = 'tmpfile-%s' % (uuid.uuid4())
+    full_filename = os.path.join(tempdir, basename)
+    open(full_filename, 'w').close()
```

Comment:
> open(full_filename, 'w').close()

Seems we are doing an `open` followed by a `close` to create an empty file in this line.  I guess we still need to do the empty temp file creation even with the subsequent line?

>with open(full_filename, mode) as f:

Seems `mode` is set to `'w'` when we call `temporary_file` in `app.py` which I believe would create the file.